src/components/shared/WindowTitle.js

```diff
@@ -0,0 +1,121 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import { PureComponent } from 'react';
+import explicitConnect from '../../utils/connect';
+
+import { getProfileName } from '../../selectors/url-state';
+import { getProfile } from '../../selectors/profile';
+
+import type { Profile, ProfileMeta } from '../../types/profile';
+import type {
+  ExplicitConnectOptions,
+  ConnectedProps,
+} from '../../utils/connect';
+
+type StateProps = {|
+  +profile: Profile,
+  +profileName: string | null,
+|};
+
+type Props = ConnectedProps<{||}, StateProps, {||}>;
+
+class WindowTitle extends PureComponent<Props> {
+  // This component updates window title in the form of:
+  // profile name - version - platform - date time - 'perf.html'
+  componentDidUpdate() {
+    const { profile, profileName } = this.props;
+    const { meta } = profile;
+    let title = '';
+
+    if (profileName) {
+      title = title.concat(profileName, ' - ');
+    }
+    if (meta) {
+      title = title.concat(_formatVersion(meta), ' - ');
+    }
+    if (meta && meta.oscpu) {
+      title = title.concat(_formatPlatform(meta), ' - ');
+    }
+    if (meta && meta.startTime) {
+      title = title.concat(_formatDateTime(meta.startTime), ' - ');
+    }
+    title = title.concat('perf.html');
```

Comment:
note: we renamed the project since we wrote it in the profile, you can use `Firefox profiler` instead.